src/App.jsx

```diff
@@ -14,33 +17,91 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const RESET_SORTING = 0;
+const ALPHABETICAL_SORTING = 1;
+const SORT_BY_LENGTH = 2;
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function prepareGoods(goods, sortType, reversed) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  preparedGoods.sort((goodA, goodB) => {
+    switch (sortType) {
+      case ALPHABETICAL_SORTING:
+        return goodA.localeCompare(goodB);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      case SORT_BY_LENGTH:
+        return goodA.length - goodB.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return reversed ? preparedGoods.reverse() : preparedGoods;
+}
+
+export const App = () => {
+  const [reversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(RESET_SORTING);
+
+  const preparedGoods = prepareGoods(goodsFromServer, sortType, reversed);
```

Comment:
The `prepareGoods` function is called with the current `sortType` and `reversed` state. Ensure that the logic within `prepareGoods` correctly handles all sorting types, including the default case which corresponds to `RESET_SORTING`.